pyne/alara.py

```diff
@@ -234,8 +234,10 @@ def photon_source_hdf5_to_mesh(mesh, filename, tags, sub_voxel=False,
     phtn_src_dc = []
     with tb.open_file(filename) as h5f:
         for row in h5f.root.data:
-            phtn_src_dc.append(row[2])
-    phtn_src_dc = list(set(phtn_src_dc))
+            if row[2] not in phtn_src_dc:
```

Comment:
Please add a comment that explains why this is correct, particularly what the known/assumed order of the rows in the table are.

Reply:
The content in the `h5f.root.data` is a table with data type of 
```
dt = np.dtype([
        ('idx', np.int64),
        ('nuc', 'S6'),
        ('time', 'S20'),
        ('phtn_src', np.float64, G),
    ])
```
The `time` changes fastest. And for every (`idx`, `nuc`) pair, it has the exactly the same decay times. The size of the table if `number of voxels` \* `average number of nuclides in each voxel` \* `number of decay times`.

If we want to get the decay times, we don't need to iterate the entire table, what we need is to iterate the first (`idx`, `nuc`) pair and then we get decay times.

This PR reduces the iterate time form `number of voxels` \* `average number of nuclides in each voxel` \* `number of decay times` to `number of decay times` + 1.